KMW / 7578 Most Engaging Pages KMW Tile

10upsimon opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #7578

## Relevant technical choices

- Defined new `MostEngagingPagesWidget` component and subsequent constants and identifiers such as `KM_ANALYTICS_MOST_ENGAGING_PAGES`.
- Registered widget in GA4 module index, behind `newsKeyMetrics` feature flag.
- Added `MostEngagingPagesWidget` to list of key metrics widgets for side panel display.
- Added applicable VRT tests.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

10upsimon commented:
Note to reviewer: The 2x failing e2e tests appear to be as a result of current known test issues.

nfmohit commented on assets/js/modules/analytics-4/components/widgets/MostEngagingPagesWidget.stories.js:
```diff
@@ -0,0 +1,342 @@
+	</WithRegistrySetup>
+);
+
+export const Ready = Template.bind( {} );
```
I think all the existing metric tiles with the tile table format have a "Ready View Only" story as well. Do you think it's worth adding that here as well?

10upsimon replied:
Good catch @nfmohit , I will add said story and request a re-review

10upsimon replied:
@nfmohit I have addressed the above, please see https://github.com/google/site-kit-wp/pull/7660#issuecomment-1750126935

nfmohit reviewed (changes requested):
Hi @10upsimon! This is looking solid, thank you!

I have left a minor comment but otherwise, this should be good to go.

10upsimon commented:
@nfmohit I have addressed the following issues:

1. Added a story scenario for "Ready view only", addressed in https://github.com/google/site-kit-wp/pull/7660/commits/8b552a54f16eee9e2f3f603f7631eada0659c585
2. Updated VRT reference images IRO the above, addressed in https://github.com/google/site-kit-wp/pull/7660/commits/5a5707aba1a579bfbe7c0d272fe3921833e967c3

Requesting a re-review, thank you.

nfmohit reviewed (approved):
LGTM, thank you @10upsimon !

eugene-manuilov commented on assets/js/modules/analytics-4/components/widgets/MostEngagingPagesWidget.js:
```diff
@@ -0,0 +1,240 @@
+		} )
+	);
+
+	const pageViewsReportOptions = {
+		...dates,
+		dimensions: [ 'pagePath' ],
+		metrics: [ { name: 'screenPageViews' } ],
+	};
```
Let's limit the number of rows selected by this report to 1 to avoid fetching too much data. If we don't set the limit, the report will select all rows matching provided criteria.

eugene-manuilov commented on assets/js/modules/analytics-4/components/widgets/MostEngagingPagesWidget.js:
```diff
@@ -0,0 +1,240 @@
+		] )
+	);
+
+	const report = useSelect( ( select ) => {
```
We should use the `useInViewSelect` hook to load this report only when this widget gets into the viewport.

eugene-manuilov reviewed (changes requested):
Nice work, @10upsimon. Two more comments from me and this one is ready to go 👍

eugene-manuilov reviewed (approved):
Thanks, @10upsimon and @jimmymadon. LGTM.
